pkg/expression/helper.go

```diff
@@ -73,7 +73,7 @@ func getTimeCurrentTimeStamp(ctx EvalContext, tp byte, fsp int) (t types.Time, e
 	}
 	value.SetCoreTime(types.FromGoTime(defaultTime.Truncate(time.Duration(math.Pow10(9-fsp)) * time.Nanosecond)))
 	if tp == mysql.TypeTimestamp || tp == mysql.TypeDatetime || tp == mysql.TypeDate {
-		err = value.ConvertTimeZone(time.Local, ctx.Location())
+		err = value.ConvertTimeZone(defaultTime.Location(), ctx.Location())
```

Comment:
This is the only line of the logic change in the production code.

When calling `ConvertTimeZone`, we use the timezone from `defaultTime` instead of local timezone to make sure the result is correct even if `getStmtTimestamp` returns a time that is not in `Local` timezone.